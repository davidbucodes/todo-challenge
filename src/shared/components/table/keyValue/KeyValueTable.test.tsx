import { composeStories } from "@storybook/react";
import { render, screen } from "@testing-library/react";
import * as stories from "./KeyValueTable.stories";

const { Default, Loading, Error, NoDataProvided } = composeStories(stories);

describe("KeyValueTable", () => {
  describe("Default", () => {
    it("should render correctly", () => {
      const { container } = render(<Default />);
      expect(container).toMatchSnapshot();
      const column = screen.getByText(/item number:/i);
      const value = screen.getByText(/1/i);
      expect(column).toBeInTheDocument();
      expect(value).toBeInTheDocument();
    });
  });

  describe("Loading", () => {
    it("should render correctly", () => {
      const { container } = render(<Loading />);
      expect(container).toMatchSnapshot();
      const values = screen.getAllByText(/Loading data.../i);
      expect(values).toHaveLength(2);
    });
  });

  describe("Error", () => {
    it("should render correctly", () => {
      const { container } = render(<Error />);
      expect(container).toMatchSnapshot();
      const values = screen.getAllByText(/Error loading data/i);
      expect(values).toHaveLength(2);
    });
  });

  describe("NoDataProvided", () => {
    it("should render correctly", () => {
      const { container } = render(<NoDataProvided />);
      expect(container).toMatchSnapshot();
      const values = screen.getAllByText(/No data provided/i);
      expect(values).toHaveLength(2);
    });
  });
});
