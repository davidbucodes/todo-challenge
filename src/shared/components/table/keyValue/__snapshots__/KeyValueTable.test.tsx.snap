// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValueTable Default should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi"
    >
      <div
        class="sc-ewnqHT cJqMKh"
      >
        <span
          class="sc-fFGjHI bkirf"
        >
          item number
          :
        </span>
        <span
          class="sc-dicizt jiIufD"
        >
          1
        </span>
      </div>
      <div
        class="sc-ewnqHT cJqMKh"
      >
        <span
          class="sc-fFGjHI bkirf"
        >
          title
          :
        </span>
        <span
          class="sc-dicizt jiIufD"
        >
          delectus aut autem
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`KeyValueTable Error should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi"
    >
      <div
        class="sc-ewnqHT cJqMKh"
      >
        <span
          class="sc-fFGjHI bkirf"
        >
          item number
          :
        </span>
        <span
          class="sc-dicizt jiIufD"
        >
          Error loading data
        </span>
      </div>
      <div
        class="sc-ewnqHT cJqMKh"
      >
        <span
          class="sc-fFGjHI bkirf"
        >
          title
          :
        </span>
        <span
          class="sc-dicizt jiIufD"
        >
          Error loading data
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`KeyValueTable Loading should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi"
    >
      <div
        class="sc-ewnqHT cJqMKh"
      >
        <span
          class="sc-fFGjHI bkirf"
        >
          item number
          :
        </span>
        <span
          class="sc-dicizt jiIufD"
        >
          Loading data...
        </span>
      </div>
      <div
        class="sc-ewnqHT cJqMKh"
      >
        <span
          class="sc-fFGjHI bkirf"
        >
          title
          :
        </span>
        <span
          class="sc-dicizt jiIufD"
        >
          Loading data...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`KeyValueTable NoDataProvided should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi"
    >
      <div
        class="sc-ewnqHT cJqMKh"
      >
        <span
          class="sc-fFGjHI bkirf"
        >
          item number
          :
        </span>
        <span
          class="sc-dicizt jiIufD"
        >
          No data provided
        </span>
      </div>
      <div
        class="sc-ewnqHT cJqMKh"
      >
        <span
          class="sc-fFGjHI bkirf"
        >
          title
          :
        </span>
        <span
          class="sc-dicizt jiIufD"
        >
          No data provided
        </span>
      </div>
    </div>
  </div>
</div>
`;
